### DISCUSSION TITLE
Proxy Driver

### DISCUSSION THREAD

**Discussion by cb-rajeshkumar** [serenity-bdd/serenity-core#3079] [General]
Can some help me launch proxy driver. I have below code.

ChromeDriver driver = (ChromeDriver) driverFacade.getProxiedDriver();
        DevTools devTools = driver.getDevTools();
        devTools.createSession();
        devTools.send(Network.enable());

        File harFile = new File("intercepted-requests.har");
        try (FileOutputStream fos = new FileOutputStream(harFile)) {
            devTools.send(Network.enable(new Network.EnableCommandConfig().setAttachRequestId(true)
                    .setCaptureRequestHeaders(true).setCaptureResponseHeaders(true)));
            devTools.addListener(Network.requestWillBeSent(), request -> {
                try {
                    fos.write(request.getRequestId().getBytes());
                    fos.write(request.getRequest().getUrl().getBytes());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            });

            actor.attemptsTo(Open.url("https://www.google.com"));
        } catch (IOException e) {
            e.printStackTrace();
        }

Seems like driverFacade class is not available in latest serenity version

**Comment by Deepak3991** [comment]
@globalworming @wakaleo   Is there any sample project to use [serenity-browsermob-plugin](https://github.com/serenity-bdd/serenity-core/tree/main/serenity-browsermob-plugin)? I am using older version of serenity below 3 in one of my projects. Now the ask is to capture network traffic. I saw serenity does have BrowserMobProxy kind of option called browsermob-plugin. But i am not able to find any working sample here. Can anyone please suggest. Thanks

**Comment by globalworming** [comment]
first time i hear about this..

what do you actually want to achieve here? what problem are you trying to solve?

**Comment by Deepak3991** [comment]
Thanks for your quick response @globalworming 
In order to perform isolated UI component testing with mock server, I am trying to capture network traffic logs from browser. Application under test is designed in ReactJs. so soon as I perform any event like pagination, filter, sorting, or click on update button it sends JSON request to the backend servers, this is what I need to capture & validate like its response status and all. 
Post Serenity 3, there is a class called DevTools, this does the same thing but since my project is not yet migrated to serenity 3, I don't have luxury to use that class.
Hoping I explanation help you.
